Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp

```diff
@@ -56,20 +56,46 @@
 #include <wtf/text/AtomStringHash.h>
 #include <wtf/text/StringToIntegerConversion.h>
 
+static void videoDecoderLimitsDefaults(unsigned& mediaMaxWidth, unsigned& mediaMaxHeight, unsigned& mediaMaxFrameRate)
+{
+#ifdef VIDEO_DECODING_LIMIT
+    // VIDEO_DECODING_LIMIT should be in format: WIDTHxHEIGHT@FRAMERATE
+    String videoDecodingLimit(String::fromUTF8(VIDEO_DECODING_LIMIT));
+
+    if (videoDecodingLimit.isEmpty())
+        return;
+
+    Vector<String> entries;
+
+    // extract frame rate part from the VIDEO_DECODING_LIMIT: WIDTHxHEIGHT@FRAMERATE
+    videoDecodingLimit.split('@', [&entries](StringView item) {
+        entries.append(item.toString());
+    });
+
+    if (entries.size() != 2)
+        return;
+
+    mediaMaxFrameRate = parseIntegerAllowingTrailingJunk<unsigned>(entries[1]).value_or(0);
+
+    String widthAndHeight = entries[0];
+    entries.clear();
+
+    // extract WIDTH and HEIGHT from: WIDTHxHEIGHT
+    widthAndHeight.split('x', [&entries](StringView item) {
+        entries.append(item.toString());
+    });
+
+    if (entries.size() != 2)
+        return;
+
+    mediaMaxWidth = parseIntegerAllowingTrailingJunk<unsigned>(entries[0]).value_or(0);
```

Comment:
Instead of `value_or()`, use `!has_value()` to early return false when the optional value isn't set.